iaso/api/fhir/location.py

```diff
@@ -0,0 +1,304 @@
+            prev_params = request.query_params.copy()
+            prev_params["_skip"] = max(0, offset - page_size)
+            prev_url = f"{request.build_absolute_uri().split('?')[0]}?{prev_params.urlencode()}"
+            links.append({"relation": "previous", "url": prev_url})
+
+        return links
+
+    def list(self, request, *args, **kwargs):
+        """
+        List all Location resources as FHIR Bundle
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+
+        # Apply pagination manually to get total count
+        total_count = queryset.count()
+        page_size, offset = self._get_pagination_params(request)
+
+        paginated_queryset = queryset[offset : offset + page_size]
+
+        # Serialize locations
+        serializer = self.get_serializer(paginated_queryset, many=True)
+
+        # Create FHIR Bundle
+        bundle = self.create_fhir_bundle(serializer.data, total_count, request)
+
+        return Response(bundle)
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieve a single Location resource by ID
+        """
+        try:
+            instance = self.get_object()
+            serializer = self.get_serializer(instance)
+            return Response(serializer.data)
+        except Http404:
+            error_response = {
+                "resourceType": "OperationOutcome",
+                "issue": [
+                    {
+                        "severity": "error",
+                        "code": "not-found",
+                        "details": {"text": f"Location with id '{kwargs.get('pk')}' not found"},
+                    }
+                ],
+            }
+            return Response(error_response, status=status.HTTP_404_NOT_FOUND)
+
+    @action(detail=True, methods=["get"])
+    def children(self, request, pk=None):
+        """
+        Get child locations (sub org units) of a specific location
+        """
+        try:
+            parent_org_unit = self.get_object()
+            children_queryset = self.get_queryset().children(parent_org_unit)
+
+            # Serialize child locations
+            serializer = self.get_serializer(children_queryset, many=True)
+
+            # Create FHIR Bundle for children
+            bundle = self.create_fhir_bundle(serializer.data, children_queryset.count(), request)
+            bundle["id"] = f"children-{pk}"
+
+            return Response(bundle)
+
+        except Http404:
+            error_response = {
+                "resourceType": "OperationOutcome",
+                "issue": [
+                    {
+                        "severity": "error",
+                        "code": "not-found",
+                        "details": {"text": f"Location with id '{pk}' not found"},
+                    }
+                ],
+            }
+            return Response(error_response, status=status.HTTP_404_NOT_FOUND)
+
+    @action(detail=False, methods=["get"])
+    def metadata(self, request):
```

Comment:
Is getting `metadata` via a `GET` method a FHIR API requirement?

I'm asking because there is an existing [metadata mechanism in DRF](https://www.django-rest-framework.org/api-guide/metadata/) but it relies on the `OPTIONS` HTTP method.